src/mame/cinematronics/cinemat.cpp

```diff
@@ -353,38 +353,60 @@ static INPUT_PORTS_START( spacewar )
 	PORT_BIT( 0x0040, IP_ACTIVE_LOW, IPT_OTHER ) PORT_NAME("Option 2") PORT_CODE(KEYCODE_2_PAD)
 	PORT_BIT( 0x0080, IP_ACTIVE_LOW, IPT_OTHER ) PORT_NAME("Option 7") PORT_CODE(KEYCODE_7_PAD)
 	PORT_BIT( 0x0100, IP_ACTIVE_LOW, IPT_JOYSTICK_LEFT ) PORT_2WAY PORT_PLAYER(1)
-	PORT_BIT( 0x0200, IP_ACTIVE_LOW, IPT_BUTTON2 ) PORT_PLAYER(2)
+	PORT_BIT( 0x0200, IP_ACTIVE_LOW, IPT_BUTTON2 ) PORT_PLAYER(2) PORT_NAME("Forward")
 	PORT_BIT( 0x0400, IP_ACTIVE_LOW, IPT_OTHER ) PORT_NAME("Option 5") PORT_CODE(KEYCODE_5_PAD)
 	PORT_BIT( 0x0800, IP_ACTIVE_LOW, IPT_OTHER ) PORT_NAME("Option 0") PORT_CODE(KEYCODE_0_PAD)
 	PORT_BIT( 0x1000, IP_ACTIVE_LOW, IPT_JOYSTICK_RIGHT ) PORT_2WAY PORT_PLAYER(2)
 	PORT_BIT( 0x2000, IP_ACTIVE_LOW, IPT_JOYSTICK_RIGHT ) PORT_2WAY PORT_PLAYER(1)
 	PORT_BIT( 0x4000, IP_ACTIVE_LOW, IPT_JOYSTICK_LEFT ) PORT_2WAY PORT_PLAYER(2)
-	PORT_BIT( 0x8000, IP_ACTIVE_LOW, IPT_BUTTON2 ) PORT_PLAYER(1)
+	PORT_BIT( 0x8000, IP_ACTIVE_LOW, IPT_BUTTON2 ) PORT_PLAYER(1) PORT_NAME("Forward")
 
 	PORT_START("SWITCHES")
 	PORT_DIPNAME( 0x03, 0x00,  "Time" )
 	PORT_DIPSETTING(    0x03, "0:45/coin" )
 	PORT_DIPSETTING(    0x00, "1:00/coin" )
 	PORT_DIPSETTING(    0x01, "1:30/coin" )
 	PORT_DIPSETTING(    0x02, "2:00/coin" )
-	PORT_BIT( 0x04, IP_ACTIVE_LOW, IPT_BUTTON1 ) PORT_PLAYER(2)
-	PORT_BIT( 0x08, IP_ACTIVE_LOW, IPT_BUTTON3 ) PORT_PLAYER(2)
-	PORT_BIT( 0x10, IP_ACTIVE_LOW, IPT_BUTTON1 ) PORT_PLAYER(1)
-	PORT_BIT( 0x20, IP_ACTIVE_LOW, IPT_BUTTON3 ) PORT_PLAYER(1)
+	PORT_BIT( 0x04, IP_ACTIVE_LOW, IPT_BUTTON1 ) PORT_PLAYER(2) PORT_NAME("Fire")
+	PORT_BIT( 0x08, IP_ACTIVE_LOW, IPT_BUTTON3 ) PORT_PLAYER(2) PORT_NAME("Hyperspace")
+	PORT_BIT( 0x10, IP_ACTIVE_LOW, IPT_BUTTON1 ) PORT_PLAYER(1) PORT_NAME("Fire")
+	PORT_BIT( 0x20, IP_ACTIVE_LOW, IPT_BUTTON3 ) PORT_PLAYER(1) PORT_NAME("Hyperspace")
```

Comment:
Same here, this is going to be confusing in menus.  The `%p` pseudo-format for localised player exists for a reason.  But that aside, in most cases, it’s fine to just leave the default button names and allow the port name plugin to be used to supply “friendly” names.

Reply:
Ah, I wasn't aware of the portname plugin. I'll remove the added labels.

Reply:
Personally, I'm ok with default PORT_NAME. I wouldn't use it when a game only has 2 buttons like Attack/Jump, but I would add them if things get confusing. For example: https://commons.wikimedia.org/wiki/File:Basketball_2_by_Mattel_Electronics,_Made_In_Hong_Kong,_Copyright_1979_%28LED_Electronic_Handheld_Game%29.jpg
Or this one with 24 buttons: https://www.schach-computer.info/wiki/images/d/d1/Novag_Diablo-1.jpg

The portname plugin is nice, but hard to maintain due to separate code/data (eg. port tags, bitmasks). Plus expecting users to find/download/update the portname files.